cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml

```diff
@@ -1,33 +1,50 @@
 name: Publish {{ cookiecutter.project_name }} to PyPI
 
 on:
-  release:
-    types: [created]
+    release:
+        types: [created]
 
 jobs:
-  deploy:
-    runs-on: ubuntu-latest
-    if: "!contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[ci skip]')"
-    steps:
-    - uses: actions/checkout@v2
-      name: Check out source-code repository
+    release:
+        name: Release
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check out the repository
+              uses: actions/checkout@v2.3.4
+              with:
+                  fetch-depth: 2
 
-    - name: Set up Python
-      uses: actions/setup-python@v1
-      with:
-        python-version: '3.8'
+            - name: Set up Python
+              uses: actions/setup-python@v2.1.4
+              with:
+                  python-version: "3.9"
 
-    - name: Install pip, setuptools, wheel, twine
-      run: |
-        python -m pip install --upgrade pip
-        pip install setuptools wheel twine
+            - name: Upgrade pip
+              run: |
+                  pip install --constraint=.github/workflows/constraints.txt pip
+                  pip --version
 
-    - name: Build and publish
-      env:
-        TWINE_USERNAME: {% raw %}${{ secrets.PYPI_USERNAME }}{% endraw %}
-        # TODO COOKIETEMPLE: Configure your secrets to enable automatic deployment to PyPi on releases
-        # https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets
-        TWINE_PASSWORD: {% raw %}${{ secrets.PYPI_PASSWORD }}{% endraw %}
-      run: |
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
+            - name: Install Poetry
+              run: |
+                  pip install --constraint=.github/workflows/constraints.txt poetry
+                  poetry --version
```

Comment:
this is the same as pip above, could this be somehow combined?

Reply:
I would say no, since pip is special in that it is being used to install poetry and to install poetry we must ensure to use the latest pip version